test/test-functions

```diff
@@ -84,6 +84,26 @@ if [[ "$IS_BUILT_WITH_ASAN" = "yes" ]]; then
     fi
 fi
 
+function should_use_kvm() {
+    # TEST_NO_KVM is set: don't use KVM
+    [ -z "$TEST_NO_KVM" ] || return 1
+
+    # no /dev/kvm: can't use KVM
+    [[ -c /dev/kvm ]] || return 1
+
+    if [[ $(systemd-detect-virt -v) = kvm ]]; then
+        [ "$ARCH" ] || ARCH=$(uname -m)
+        # if running under x86 KVM, nested KVM is ok
```

Comment:
https://bugs.launchpad.net/ubuntu/+source/linux-lts-xenial/+bug/1682077 is still open so I'm not sure how safe it is to bring nested kvm back.

Reply:
ok, i just forced-pushed a new version, where it does not enable nested kvm for any archs, unless a variable specifically tells it to, either $TEST_NESTED_KVM or $TEST_NESTED_KVM_$ARCH (e.g. $TEST_NESTED_KVM_x86_64).  My thinking is this will allow normal users to run tests on their bare metal and kvm will be used, and normal users can also run tests in a kvm guest, and nested kvm won't be used regardless of the arch (i.e., the status quo).  Then, the Ubuntu CI webhooks can add TEST_NESTED_KVM_x86_64=1 and TEST_NESTED_KVM_i386=1 to allow nested kvm in the Ubuntu CI testbed, which seems to work (in my testing, at least)